File: pico-statsd/src/main/scala/org/pico/statsd/Tags.java
package org.pico.statsd;

public class Tags {
    /**
     * Generate a suffix conveying the given tag list to the client
     */
    public static String tagString(final String[] tags, final String tagPrefix) {
        final StringBuilder sb;

        if (tagPrefix != null) {
            if ((tags == null) || (tags.length == 0)) {
                return tagPrefix;
            }

            sb = new StringBuilder(tagPrefix);
            sb.append(",");
        } else {
            if ((tags == null) || (tags.length == 0)) {
                return "";
            }

            sb = new StringBuilder("|#");
        }

        for (int n=tags.length - 1; n>=0; n--) {
            sb.append(tags[n]);

            if (n > 0) {
                sb.append(",");
            }
        }

        return sb.toString();
    }
}

File: pico-statsd/src/main/scala/org/pico/statsd/Tags.scala
package org.pico.statsd

object Tags {
  /**
    * Generate a suffix conveying the given tag list to the client
    */
  def tagString(tags: Seq[String], tagPrefix: String): String = {
    if (tagPrefix != null) {
      if (tags == null || tags.isEmpty) {
        tagPrefix
      } else {
        val sb = new StringBuilder(tagPrefix)
        sb.append(",")
        sb.toString
      }
    } else {
      if (tags == null || tags.isEmpty) {
        ""
      } else {
        val sb = new StringBuilder("|#")
        var n: Int = tags.length - 1

        while (n >= 0) {
          sb.append(tags(n))

          if (n > 0) {
            sb.append(",")
          }

          n -= 1
          n + 1
        }

        sb.toString
      }
    }
  }
}
